### PR TITLE
[PM-7960] Revert change from passkey provider pr

### PR DESCRIPTION
**Pull request by andrebispo5**
## Type of change
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Fix bug where users were unable to decrypt vault with MP after SSO authentication.

## Code changes

Revert change from passkey provider PR 

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/Pages/Accounts/LockPage.xaml.cs [hunk 1]
```diff
@@ -168,7 +168,7 @@ private void Unlock_Clicked(object sender, EventArgs e)
                 var tasks = Task.Run(async () =>
                 {
                     await Task.Delay(50);
-                    _vm.SubmitCommand.Execute(null);
+                    MainThread.BeginInvokeOnMainThread(async () => await _vm.SubmitAsync());
```
Comment (fedemkr) [inline src/Core/Pages/Accounts/LockPage.xaml.cs:171]: 🤔 Why is this necessary to be reverted? Do we need to invoke on main thread?

Comment (andrebispo5) [reply src/Core/Pages/Accounts/LockPage.xaml.cs:171]: This was the only change in that flow and reverting it fixed the issue, also the AppCenter report and my bug reproduction was referring a invocation out of main thread issue.

Comment (fedemkr) [reply src/Core/Pages/Accounts/LockPage.xaml.cs:171]: @andrebispo5 could you check if instead of doing this you change the command on the VM to be like this:
```c#
SubmitCommand = CreateDefaultAsyncRelayCommand(() => MainThread.InvokeOnMainThreadAsync(SubmitAsync), onException: _logger.Exception, allowsMultipleExecutions: false);
```
Because there was an issue on the passkeys branch where  in some occasions the submit flow was being called multiple times (basically the `DoOnce` was not taking the appropriate effect), here you have [the commit](https://github.com/bitwarden/mobile/pull/3020/commits/7fded63a703a98b3aa7fe7600d67923bd4698ddb).
This was happening on the `Return` keyboard but if I recall correctly it was in this `Unlock_Clicked` as well. 
If the VM thing doesn't work, we can leave this as is and see if we have any issue reported because of it from QA or in the future.

**Review by fedemkr — APPROVED** [review]
